File: android/app/src/main/java/com/sdust/iotroniks/ui/main/MainPresenter.java
package com.sdust.iotroniks.ui.main;

import com.sdust.iotroniks.data.AppDataManagerHelper;
import com.sdust.iotroniks.data.DataManagerHelper;
import com.sdust.iotroniks.data.Incident;

import java.util.List;

class MainPresenter implements MainContract.Presenter {

  private static final String TAG = "MainPresenter";

  private final MainContract.View mainView;
  private final AppDataManagerHelper appDataManagerHelper;

  MainPresenter(AppDataManagerHelper appDataManagerHelper, MainContract.View mainView) {
    this.mainView = mainView;
    this.mainView.setPresenter(this);
    this.appDataManagerHelper = appDataManagerHelper;
  }

  @Override
  public void start() {
    mainView.showNoIncidents();
    loadIncidents();
  }

  @Override
  public void result(int requestCode, int resultCode) {

  }

  @Override
  public void loadIncidents() {
    appDataManagerHelper.getIncidents(new DataManagerHelper.LoadIncidentsCallback() {
      @Override
      public void onIncidentsLoaded(List<Incident> incidents) {
        mainView.showIncidents(incidents);
      }
    });
  }
}

File: android/app/src/main/java/com/sdust/iotroniks/ui/main/MainPresenter.kt
package com.sdust.iotroniks.ui.main

import com.sdust.iotroniks.data.AppDataManagerHelper
import com.sdust.iotroniks.data.DataManagerHelper
import com.sdust.iotroniks.data.Incident

internal class MainPresenter(private val appDataManagerHelper: AppDataManagerHelper, private val mainView: MainContract.View) : MainContract.Presenter {

  init {
    this.mainView.setPresenter(this)
  }

  override fun start() {
    mainView.showNoIncidents()
    loadIncidents()
  }

  override fun result(requestCode: Int, resultCode: Int) {

  }

  override fun loadIncidents() {
    appDataManagerHelper.getIncidents(object : DataManagerHelper.LoadIncidentsCallback {
      override fun onIncidentsLoaded(incidentList: List<Incident>) {
        mainView.showIncidents(incidentList)
      }
    })
  }

  companion object {

    private val TAG = "MainPresenter"
  }
}
